node/consensus/src/lib.rs

```diff
@@ -389,15 +489,13 @@ impl<N: Network> Consensus<N> {
                     .lock()
                     .insert(TransmissionID::Transaction(transaction.id(), checksum), timestamp);
             }
-            // Add the transaction to the memory pool.
-            trace!("Received unconfirmed transaction '{}' in the queue", fmt_id(transaction_id));
-            if transaction.is_deploy() {
-                if self.transactions_queue.lock().deployments.put(transaction_id, transaction).is_some() {
-                    bail!("Transaction '{}' exists in the memory pool", fmt_id(transaction_id));
-                }
-            } else if self.transactions_queue.lock().executions.put(transaction_id, transaction).is_some() {
+            // Check that the transaction is not in the mempool.
+            if self.contains_transaction(&transaction_id) {
```

Comment:
Any chance we can get rid of contains, by letting insert return a boolean if it was already present? Similar to the pre-existing behaviour for TransactionsQueue? That way we only have to lock once.

Reply:
> I see! Should I update the implementation in this PR to allow overwriting transactions in the mempool (I removed it thinking it was not desired)?

It's probably fine, but you can still avoid it by 1. taking the lock 2. calling `contains_key` 3. if it doesn't contain, then insert it.

A related nit: executions are more common than deployments, so if you want to short_circuit, try executions first, or just use `is_deploy` like below: https://github.com/ProvableHQ/snarkOS/pull/3697/files#diff-8b822371aae41d2275688c666d4c5d5d304e67aa422628ee42c405b5e8d7f572R162